.github/workflows/conformance-externalworkloads-v1.10.yaml

```diff
@@ -313,10 +313,17 @@ jobs:
 
       - name: Clean up GKE cluster and VM
         if: ${{ always() }}
-        run: |
-          gcloud container clusters delete ${{ env.clusterName }} --zone ${{ env.zone }} --quiet --async
-          gcloud compute instances delete ${{ env.vmName }} --zone ${{ env.zone }} --quiet
-        shell: bash {0} # Disable default fail-fast behaviour so that all commands run independently
+        uses: nick-invision/retry@3e91a01664abd3c5cd539100d10d33b9c5b68482
```

Comment:
What do you think of using a Bash loop to check against ongoing operations with `gcloud container operations list` before trying to delete, instead of using a `step`-level retry? I think your solution works as-is, but I feel like it would be more resilient since it would not rely on a fixed number of retries.

FYI we have a working example of that in the Jenkins GKE cleanup step: https://github.com/cilium/cilium/blob/b332f4b8ff4e2cdceddeff93888127820ee1241b/test/gke/release-cluster.sh#L18-L25

This snippet has the Jenkins cleanup wait for no all ongoing operations to be finished before trying to delete the cluster.

Reply:
I thought about using a bash script since it is harder to control the flow in a inline run job. I have even written one similar to the suggested version. There wasa couple of reasons I thought this approach might be better. 

1. Readability. It is much easer to read, and follow a workflow if you know what a job does instead of jumping to another file to see.
2. Consistency. It is not exactly same usage but I saw re-try loop used another workflow. But then again it is not that important, I guess.
3. I did not want to put a infinite loop without a timeout. But I guess workflow has a timeout so that take care of that.

My only concern is still readability. Bu I have to admit the script is a bit ugly 😄 

If you still think best way to go is script, just let me know.

Reply:
I have updated to code as we talked about. One thing that might be little hard to spot at first glance is that I use regex `~` for matching clusters in filter line. This allows us to catch the multiple clusters at once. This is especially useful in multicluster test.